functions/New-DbaLinkedServer.ps1

```diff
@@ -0,0 +1,144 @@
+
+        Creates a new linked server named linkedServer1 on the sql01 instance. The link is via the SQL Native Client and is connected to the sql02 instance.
+
+    .EXAMPLE
+        PS C:\>Connect-DbaInstance -SqlInstance sql01 | New-DbaLinkedServer -LinkedServer linkedServer1 -ServerProduct mssql -Provider sqlncli -DataSource sql02
+
+        Creates a new linked server named linkedServer1 on the sql01 instance. The link is via the SQL Native Client and is connected to the sql02 instance. The sql01 instance is passed in via pipeline.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string]$LinkedServer,
+        [string]$ServerProduct,
+        [string]$Provider,
+        [string]$DataSource,
+        [string]$Location,
+        [string]$ProviderString,
+        [string]$Catalog,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Server[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+
+        if (Test-Bound -Not -ParameterName LinkedServer) {
+            Stop-Function -Message "LinkedServer is required"
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            $InputObject += Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+        }
+
+        foreach ($instance in $InputObject) {
+
+            if ($instance.LinkedServers.Name -contains $LinkedServer) {
+                Stop-Function -Message "Linked server $LinkedServer already exists on $($instance.Name)" -Continue
+            }
+
+            if ($Pscmdlet.ShouldProcess($instance.Name, "Creating the linked server $LinkedServer on $($instance.Name)")) {
+                try {
+                    $newLinkedServer = New-Object Microsoft.SqlServer.Management.Smo.LinkedServer -ArgumentList $instance, $LinkedServer
+
+                    if (Test-Bound ServerProduct) {
+                        $newLinkedServer.ProductName = $ServerProduct
+                    }
+
+                    if (Test-Bound Provider) {
+                        $newLinkedServer.ProviderName = $Provider
+                    }
+
+                    if (Test-Bound DataSource) {
+                        $newLinkedServer.DataSource = $DataSource
+                    }
+
+                    if (Test-Bound Location) {
+                        $newLinkedServer.Location = $Location
+                    }
+
+                    if (Test-Bound ProviderString) {
+                        $newLinkedServer.ProviderString = $ProviderString
+                    }
+
+                    if (Test-Bound Catalog) {
+                        $newLinkedServer.Catalog = $Catalog
+                    }
+
+                    $newLinkedServer.Create()
+                    $newLinkedServer
```

Comment:
Please use Get-DbaLinkedServer to get a nice object with the custom properties for the new linked server:
```
$instance |  Get-DbaLinkedServer -LinkedServer $LinkedServer
```